pkg/networking/backend_sg_provider.go

```diff
@@ -40,9 +43,10 @@ const (
 // BackendSGProvider is responsible for providing backend security groups
 type BackendSGProvider interface {
 	// Get returns the backend security group to use
-	Get(ctx context.Context) (string, error)
+	Get(ctx context.Context, activeResources []metav1.Object) (string, error)
 	// Release cleans up the auto-generated backend SG if necessary
-	Release(ctx context.Context) error
+	Release(ctx context.Context, activeResources []metav1.Object,
```

Comment:
why we do need this `activeResources` and `backendSGRequired` in release
is it possible that we just have 
```
Allocate(ctx context.Context, resources []metav1.Object)
Release(ctx context.Context, resources []metav1.Object)
```
And during release, if all tracked resource is released, and we have invoked reconcile for all resources at least once. then we release the sg?

Reply:
looks cleaner, refactored the APIs